### DISCUSSION TITLE
Adding a separate field

### DISCUSSION THREAD

**Discussion by TamaBaka** [MatthiasValvekens/pyHanko#329] [General]
On my pdf document, I have a signature field and a date field in two different places on the same page.  

I successfully managed to sign twice, with a text stamp of the signer's name in one box and a text stamp of the current date in the other box, but I get the impression that that is not how this is supposed to be done.  

Assuming that I'm not wrong, is there a way for me to render text in two different places on the page with one signing?  I checked the documentation several times but was unable to find anything that might work.  Was I supposed to use a separate pdf library to append the date field?

If, however, signing twice is perfectly valid, then I'll just go with that since it already works.

**Comment by MatthiasValvekens** [comment]
The best way to do this is to fill the date field first, and then sign in the signature field. But pyHanko currently doesn't have an API for populating non-signature fields. It's a long-requested feature though (see #35) that I haven't gotten to for lack of bandwidth and other priorities. PRs welcome; it's probably not terribly hard, but getting it right requires some PDF chops.

The way you did it probably technically produces an invalid PDF (the field value of that date field is probably supposed to be a string, not a signature), but depending on what you want to use it for that may or may not be an issue.

**Reply by TamaBaka** [reply]
Yup, lots and lots to do these days.  I had to ask because I was already trying to implement that first suggestion.  But the appended text string on the intermediate document was coming out rotated 90 degrees and mirrored, so no point signing that.  Now that I know that it's probably my best bet right now for the near-term, I can figure it out from there.  Thank you again!
